lxc/image.go

```diff
@@ -109,17 +109,40 @@ hash or alias name (if one is set).`))
 	return cmd
 }
 
-func (c *cmdImage) dereferenceAlias(d lxd.ImageServer, imageType string, inName string) string {
+func (c *cmdImage) dereferenceAlias(d lxd.ImageServer, imageType string, inName string) (string, error) {
 	if inName == "" {
 		inName = "default"
 	}
 
-	result, _, _ := d.GetImageAliasType(imageType, inName)
-	if result == nil {
-		return inName
+	result, _, err := d.GetImageAliasType(imageType, inName)
+	if err != nil {
+		// Maybe that inName is a fingerprint and can't be found as an alias
+		fingerprints, errFingerprint := d.GetImageFingerprints()
```

Comment:
Why dont we call d.GetImage() here, which as far as I can tell already supports expanding partial fingerprints too.

Why is it implemented again here?